includes/class-wc-payments-order-service.php

```diff
@@ -520,6 +534,56 @@ public function get_charge_id_for_order( $order ) : string {
 		return $order->get_meta( self::CHARGE_ID_META_KEY, true );
 	}
 
+	/**
+	 * Set the payment metadata for dispute id.
+	 *
+	 * @param  mixed  $order The order.
+	 * @param  string $dispute_id The value to be set.
+	 *
+	 * @throws Order_Not_Found_Exception
+	 */
+	public function set_dispute_id_for_order( $order, $dispute_id ) {
+		$order = $this->get_order( $order );
+		$order->update_meta_data( self::DISPUTE_ID_META_KEY, $dispute_id );
+		$order->save_meta_data();
+	}
+
+	/**
+	 * Get the payment metadata for dispute id if it exists.
+	 *
+	 * If the dispute id is not stored in order meta,
+	 * check for a dispute id via API request for disputes
+	 * created before the dispute id was stored in order meta.
+	 *
+	 * @param  mixed $order The order Id or order object.
+	 *
+	 * @return string|null
+	 *
+	 * @throws Order_Not_Found_Exception
+	 * @throws API_Exception
+	 */
+	public function get_dispute_id_for_order( $order ) : ?string {
+		$order = $this->get_order( $order );
+
+		$dispute_id = $order->get_meta( self::DISPUTE_ID_META_KEY, true );
+		if ( ! empty( $dispute_id ) ) {
+			return $dispute_id;
+		}
+
+		// For disputes created before the dispute id was stored in order meta,
+		// get the dispute id via API.
+		$response = $this->api_client->get_disputes(
+			[
+				'pagesize'        => 1,
+				'order_number_is' => $order->get_id(),
+			]
+		);
+
+		// If a dispute exists, return the dispute id, otherwise return null.
+		$dispute_id = $response['data'][0]['dispute_id'] ?? null;
```

Comment:
Does this need error handling? That's a deep reference!

Reply:
I think the null coalescing operator `??` handles that. When any of the reference chain fails, it will return null.